dbt_semantic_interfaces/implementations/semantic_manifest.py

```diff
@@ -2,15 +2,22 @@
 
 from importlib_metadata import version
 from pydantic import validator
+from typing_extensions import override
 
 from dbt_semantic_interfaces.implementations.base import HashableBaseModel
 from dbt_semantic_interfaces.implementations.metric import PydanticMetric
 from dbt_semantic_interfaces.implementations.semantic_model import PydanticSemanticModel
+from dbt_semantic_interfaces.protocols.protocol_hint import ProtocolHint
+from dbt_semantic_interfaces.protocols.semantic_manifest import SemanticManifest
 
 
-class PydanticSemanticManifest(HashableBaseModel):
+class PydanticSemanticManifest(HashableBaseModel, ProtocolHint[SemanticManifest]):
     """Model holds all the information the SemanticLayer needs to render a query."""
 
+    @override
+    def _implements_protocol(self) -> SemanticManifest:  # noqa: D
+        return self
+
```

Comment:
Oh I see, it's this approach. Yeah this is fine. I wish the toolchain just did this for us, but I can see why it's hard to do in the general case.